integrations/access/accessrequest/app.go

```diff
@@ -0,0 +1,447 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accessrequest
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/accessrequest"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/gravitational/teleport/integrations/access/common/teleport"
+	"github.com/gravitational/teleport/integrations/lib"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
+	"github.com/gravitational/teleport/integrations/lib/watcherjob"
+)
+
+const (
+	// handlerTimeout is used to bound the execution time of watcher event handler.
+	handlerTimeout = time.Second * 5
+)
+
+type App struct {
+	pluginName string
+	pluginType string
+	apiClient  teleport.Client
+	recipients common.RawRecipientsMap
+	pluginData *pd.CompareAndSwap[PluginData]
+	bot        MessagingBot
+	job        lib.ServiceJob
+}
+
+func NewApp() *App {
+	app := &App{}
+	app.job = lib.NewServiceJob(app.run)
+	return app
+}
+
+func (a *App) Init(baseApp *common.BaseApp) error {
+	a.pluginName = baseApp.PluginName
+	a.pluginType = string(baseApp.Conf.GetPluginType())
+	a.apiClient = baseApp.APIClient
+	a.recipients = baseApp.Conf.GetRecipients()
+	a.pluginData = pd.NewCAS(
+		a.apiClient,
+		a.pluginName,
+		types.KindAccessRequest,
+		EncodePluginData,
+		DecodePluginData,
+	)
+
+	var ok bool
+	a.bot, ok = baseApp.Bot.(MessagingBot)
+	if !ok {
+		return trace.BadParameter("bot does not implement access request bot methods")
+	}
+
+	return nil
+}
+
+func (a *App) Start(process *lib.Process) {
+	process.SpawnCriticalJob(a.job)
+}
+
+func (a *App) WaitReady(ctx context.Context) (bool, error) {
+	return a.job.WaitReady(ctx)
+}
+
+func (a *App) WaitForDone() {
+	<-a.job.Done()
+}
+
+func (a *App) Err() error {
+	if a.job != nil {
+		return a.job.Err()
+	}
+
+	return nil
+}
+
+func (a *App) run(ctx context.Context) error {
+	process := lib.MustGetProcess(ctx)
+
+	job, err := watcherjob.NewJob(
```

Comment:
What's the difference between this job and a.job?

Reply:
`a.job` basically wraps up the `run()` function, and WatcherJob is a pre-existing job. It'd probably be fine to remove watcherjob and migrate its functionality into this app, but for simplicity's sake I'm going to avoid that for now.